### PR TITLE
transmission_4: init at 4.0.3

### PR DESCRIPTION
**Pull request by chvp**
###### Description of changes

https://github.com/transmission/transmission/releases/tag/4.0.0
https://github.com/transmission/transmission/releases/tag/4.0.1
https://github.com/transmission/transmission/releases/tag/4.0.2
https://github.com/transmission/transmission/releases/tag/4.0.3

Previous discussion (on why this isn't just an update) in #215316.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/p2p/libutp/3.4.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, lib, fetchFromGitHub, cmake, unstableGitUpdater }:
+
+stdenv.mkDerivation rec {
+  pname = "libutp";
+  version = "unstable-2017-01-02";
```
Comment (jtojnar) [inline pkgs/applications/networking/p2p/libutp/3.4.nix:5]: This date is incorrect https://github.com/transmission/libutp/commit/9cb9f9c4f0073d78b08d6542cebaea6564ecadfe

Comment (chvp) [reply pkgs/applications/networking/p2p/libutp/3.4.nix]: Indeed, forgot to update this. Fixed now, thanks!

**Review by SuperSandro2000 — APPROVED** [review]

**Review by ambroisie — APPROVED** [review]

**Comment by chvp** [comment]
I've been using this PR instead of https://github.com/NixOS/nixpkgs/pull/215316 in my nixos-config without issues ever since creating it.

**Comment by dywedir** [comment]
Thanks!

**Comment by kirillrdy** [comment]
@chvp may I ask why nixosTest wasn't carried over from transmission 3 ? seems to be still passing for me

**Comment by chvp** [comment]
Forgetting to do so, probably.

**Pull request merged**